flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/context/SessionContext.java

```diff
@@ -111,7 +133,8 @@ public void reset() {
         // If rebuild a new Configuration, it loses control of the SessionState if users wants to
         // modify the configuration
         resetSessionConfigurationToDefault(defaultContext.getFlinkConfig());
-        this.executionContext = new ExecutionContext(executionContext);
+        buildClassLoaderAndUpdateDependencies(dependencies);
```

Comment:
Please add comment to describe why we need to rebuild the classloader here (the new classloader actually is the same as before). 

IIUC, the main reason here is because the `pipeline.jars` option has been reverted, but the `pipeline.jars` should inlcude `dependencies` jars, otherwise the udf jars can't work during runtime execution.